tests/test_swift.py

```diff
@@ -29,5 +78,100 @@ def test_frames(self):
         self.assertIsInstance(self.app.frames(), Iterable)
 
 
+class SwiftTest(unittest.TestCase):
+    """Unit test for Swift class in swift.py"""
+
+    def setUp(self):
+        """Create a QApplication and a Swift object every time."""
+        self.qapp = QApplication(sys.argv)
+        importlib.import_module = MagicMock()
+        for appInfo in APP_INFOS.values():
+            app = MagicMock()
+            app.frames.return_value = (QWidget(),)
+            cls = MagicMock(return_value=app)
+            setattr(importlib.import_module.return_value, appInfo.cls, cls)
+        self.buses = set()
+        for appInfo in APP_INFOS.values():
+            self.buses.update(appInfo.bus)
+        self.swift = Swift(APP_INFOS)
+
+    def test_init(self):
+        """Test if swift is initialized correctly."""
+        self.assertIsInstance(self.swift.mainWindow, QMainWindow)
+        self.assertIsInstance(self.swift.centralWidget, QLabel)
+        for name in APP_INFOS:
+            self.assertIn(name, self.swift._apps)
+        for bus in self.buses:
+            self.assertIn(bus, self.swift._subscribers)
```

Comment:
Umm, however, I think the purpose of testing is checking if our system works correctly and all variables should be tested even if it is protected.

What do you think of it?

Reply:
Hmm yes you are right, but for the roles of internal variables, _correct_ behavior is not _specified_.
Therefore, for me, it does not make sense to test the internal variables.
Nevertheless, if we assume something, for example an invariant condition, for some internal variables, we could test that assumption.
For internal methods or functions, I understand it because we internally specify how it should work.

Rather, I think we should define getters to iterate over the apps and subscribers.

Reply:
Okay! Then, I will remove all testing parts about protected variabls.

Reply:
For now, let's stick to this implementation.
Later I will add some public methods and modify the test codes along.